api/src/api/handlers/pda_worker.rs

```diff
@@ -0,0 +1,93 @@
+use std::str::FromStr;
+
+use crate::{
+    api::handlers::{async_verify::process_verification, is_authorized},
+    db::{
+        models::{extract_instruction, SolanaProgramBuildParams},
+        DbClient,
+    },
+    services::{get_on_chain_hash, onchain::OtterBuildParams},
+    CONFIG,
+};
+use axum::{
+    extract::State,
+    http::{HeaderMap, StatusCode},
+    Json,
+};
+use borsh::BorshDeserialize;
+use serde_json::Value;
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_sdk::pubkey::Pubkey;
+use tracing::{error, info, warn};
+
+pub(crate) async fn handle_pda_updates_creations(
+    State(db): State<DbClient>,
+    headers: HeaderMap,
+    Json(payload): Json<Vec<Value>>,
+) -> (StatusCode, &'static str) {
+    info!("Received PDA updates/creation event");
+
+    // Validate authorization
+    if !is_authorized(&headers) {
+        warn!("Unauthorized unverify attempt");
+        return (
+            StatusCode::UNAUTHORIZED,
+            "Missing or invalid authorization header",
+        );
+    }
+
+    // Validate payload
+    let instruction = match extract_instruction(&payload) {
+        Ok(instruction) => instruction,
+        Err(status) => return status,
+    };
```

Comment:
nit: can you rename this to `let helius_parsed_transaction = ...`? It's quite confusing otherwise